src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java

```diff
@@ -0,0 +1,382 @@
+package org.apache.kafka.common.security.kerberos;
+
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.security.JaasContext;
+import org.apache.kafka.common.security.JaasUtils;
+import org.apache.kafka.common.security.auth.AuthenticateCallbackHandler;
+import org.apache.kafka.common.security.authenticator.AbstractLogin;
+import org.apache.kafka.common.utils.KafkaThread;
+import org.apache.kafka.common.utils.Shell;
+import org.apache.kafka.common.utils.Time;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import javax.security.auth.Subject;
+import javax.security.auth.kerberos.KerberosPrincipal;
+import javax.security.auth.kerberos.KerberosTicket;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+/**
+ * This class is responsible for refreshing Kerberos credentials for
+ * logins for both Kafka client and server.
+ * From https://github.com/apache/kafka/blob/2181ddbb039ff688f5ff41784d943cb579f7575c/clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java
+ */
+public class KerberosLogin extends AbstractLogin {
+  private static final Logger log = LoggerFactory.getLogger(KerberosLogin.class);
+
+  private final Random RNG = new Random();
```

Comment:
Had to add this class to our codebase and convert this [previously static `Random` field](https://github.com/apache/kafka/blob/2181ddbb039ff688f5ff41784d943cb579f7575c/clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java#L53) into a non-static field.